skymap_scanner/server/utils.py

```diff
@@ -255,26 +255,41 @@ def _prevalidate(int_int_list: List[Tuple[int, int]]) -> List[Tuple[int, int]]:
     def __hash__(self) -> int:  # type: ignore[override]
         return hash(str(self._get_int_int_list()))
 
-    @cachetools.func.lru_cache()
-    def n_recos_by_nside_lowerbound(self, n_posvar: int) -> Dict[int, int]:
-        """Get # of recos per nside (w/ predictive scanning its a LOWER bound).
-
-        The actual pixel generation is done iteratively, and does not
-        rely on this function. Use this function for reporting & logging
-        only.
-        """
-
-        # override first pixel extension for the math (uses base pixel count)
-        nside_factor_list = self._get_int_int_list()
-        nside_factor_list[0] = (nside_factor_list[0][0], self.HEALPIX_BASE_PIXEL_COUNT)
-
-        def previous_nside(index: int) -> int:
-            # get previous nside value
-            if index == 0:  # for the first nside use 1, since it's used to divide
-                return 1
-            return nside_factor_list[index - 1][0]  # nside
 
-        return {
-            nside: int(n_posvar * factor * ((nside / previous_nside(i)) ** 2))
-            for i, (nside, factor) in enumerate(nside_factor_list)
-        }
+def calc_estimated_total_nside_recos(
+    nside_progression: NSideProgression,
+    reco: RecoInterface,
+    n_posvar: int,
+) -> Dict[int, int]:
+    """Get # of recos per nside (w/ predictive scanning its a LOWER bound).
+
+    The actual pixel generation is done iteratively, and does not
+    rely on this function. Use this function for reporting & logging
+    only.
+    """
+
+    def previous_nside(index: int) -> int:
+        """Get previous nside value."""
+        if index == 0:  # for the first nside use 1, since it's used to divide
+            return 1
+        return nside_progression.get_at_index(index - 1)[0]  # nside
+
+    def get_nside_count(index: int, nside: int) -> int:
+        """Get the estimated n recos for this nside."""
+        if index == 0:
+            if isinstance(reco, SplineMPEPointed):
```

Comment:
Technically this can be any pointed scan, e.g. a derived Millipede pointed class would also need this. I wonder if there's a more robust way to check based on if `pointing_dir_names is None`

Reply:
Good idea